aiida/schedulers/plugins/slurm.py

```diff
@@ -707,53 +707,68 @@ def _parse_kill_output(self, retval, stdout, stderr):
 
         return True
 
-    def parse_output(self, detailed_job_info, stdout, stderr):  # pylint: disable=inconsistent-return-statements
+    def parse_output(self, detailed_job_info=None, stdout=None, stderr=None):
         """Parse the output of the scheduler.
 
         :param detailed_job_info: dictionary with the output returned by the `Scheduler.get_detailed_job_info` command.
             This should contain the keys `retval`, `stdout` and `stderr` corresponding to the return value, stdout and
             stderr returned by the accounting command executed for a specific job id.
-        :param stdout: string with the output written by the scheduler to stdout
-        :param stderr: string with the output written by the scheduler to stderr
-        :return: None or an instance of `aiida.engine.processes.exit_code.ExitCode`
-        :raises TypeError or ValueError: if the passed arguments have incorrect type or value
+        :param stdout: string with the output written by the scheduler to stdout.
+        :param stderr: string with the output written by the scheduler to stderr.
+        :return: None or an instance of :class:`aiida.engine.processes.exit_code.ExitCode`.
+        :raises TypeError or ValueError: if the passed arguments have incorrect type or value.
```

Comment:
I guess the check on the `detailed_job_info` is scheduler specific (?).

Checks on `stdout` or `stderr` (i.e. `str` or `None`) could be done at a general level, though, if we like.

Reply:
We could, but I think it is better to keep this on the plugin level rather than collecting all possible regexes on the `Scheduler` base class.